src/resdk/resources/data.py

```diff
@@ -354,20 +357,48 @@ def download(
             "{}/{}".format(self.id, fname)
             for fname in self.files(file_name, field_name)
         ]
+        file_names = [fname for fname in self.files(file_name, field_name)]
 
-        # Only applies if downloading a single file
-        custom_file_names = None
-        if custom_file_name:
-            if file_name or field_name:
-                custom_file_names = [custom_file_name] * len(files)
-            else:
-                raise ValueError(
-                    "Setting a custom file name is not supported "
-                    "without specifying file name or field name."
-                )
+        self.resolwe._download_files(files=files, download_dir=download_dir)
+
+        return file_names
+
+    def download_and_rename(
+        self,
+        custom_file_name: str,
+        field_name: Optional[str] = None,
+        file_name: Optional[str] = None,
+        download_dir: Optional[str] = None,
+    ):
+        """Download and rename a single file from data object."""
+
+        if download_dir is None:
+            download_dir = os.getcwd()
+
+        new_file_path = os.path.join(download_dir, custom_file_name)
+
+        if os.path.exists(new_file_path):
+            logging.warning(
+                f"File with path '{new_file_path}' already exists. Skipping download."
+            )
+            return
+
+        file_names = self.download(
+            file_name=file_name,
+            field_name=field_name,
+            download_dir=download_dir,
+        )
+        if len(file_names) != 1:
+            raise ValueError(
+                f"Expected one file to be downloaded, but got {len(file_names)}"
```

Comment:
Here the files have already been downloaded and are left in the download folder. What to do with them?